### PR TITLE
Feature/initial setup dispatcher

### PR DESCRIPTION
**Pull request by ArcticWalrus**
#### What changes does this PR contain?
Sets up the structure of the service and removes the endpoints we are not planning on using.
This initial code allows the handler, model and validation to be developed by @ArtiKitten and the Game server init to be started by @ArcticWalrus 

#### Which JIRA ticket ID's are associated with this PR?
UBI-306, UBI-308

#### Does this PR change the outward facing API?
Yes

#### Additional comments and information

@ubivius/microservices

### CODE REVIEW

File: pkg/handlers/products.go [hunk 1]
```diff
@@ -2,33 +2,17 @@ package handlers
...
-// The verification of this variable is handled by gorilla/mux
-// We panic if it is not valid because that means gorilla is failing
-func getProductID(request *http.Request) int {
-	vars := mux.Vars(request)
-	id, err := strconv.Atoi(vars["id"])
-	if err != nil {
-		panic(err)
-	}
-	return id
+// NewGameHandler returns a pointer to a ProductsHandler with the logger passed as a parameter
```
Comment (MaximeStOnge) [inline pkg/handlers/products.go:15]: GameHandler à la place de ProductsHandler

File: pkg/handlers/products.go [hunk 2]
```diff
@@ -2,33 +2,17 @@ package handlers
...
-
-	"github.com/gorilla/mux"
 )
 
 // KeyProduct is a key used for the Product object inside context
 type KeyProduct struct{}
 
-// ProductsHandler contains the items common to all product handler functions
-type ProductsHandler struct {
+// GameHandler contains the items common to all product handler functions
```
Comment (MaximeStOnge) [inline pkg/handlers/products.go:10]: Game à place de product

File: pkg/handlers/products.go [hunk 3]
```diff
@@ -2,33 +2,17 @@ package handlers
 
 import (
 	"log"
-	"net/http"
-	"strconv"
-
-	"github.com/gorilla/mux"
 )
 
 // KeyProduct is a key used for the Product object inside context
```
Comment (MaximeStOnge) [inline pkg/handlers/products.go:7]: KeyGame à place de KeyProduct et Game à place de Product

File: pkg/handlers/post.go [hunk 4]
```diff
@@ -6,11 +6,11 @@ import (
 	"github.com/Ubivius/microservice-template/pkg/data"
 )
 
-// AddProduct creates a new product from the received JSON
-func (productHandler *ProductsHandler) AddProduct(responseWriter http.ResponseWriter, request *http.Request) {
-	productHandler.logger.Println("Handle POST Product")
+// NewPlayer creates a new product from the received JSON
```
Comment (MaximeStOnge) [inline pkg/handlers/post.go:9]: Changer Product pour Player ou Game

File: pkg/handlers/middleware_test.go [hunk 5]
```diff
@@ -24,11 +24,11 @@ func TestValidationMiddlewareWithValidBody(t *testing.T) {
 	request := httptest.NewRequest(http.MethodPost, "/products", strings.NewReader(string(bodyBytes)))
 	response := httptest.NewRecorder()
 
-	productHandler := NewProductsHandler(NewTestLogger())
+	productHandler := NewGameHandler(NewTestLogger())
```
Comment (MaximeStOnge) [inline pkg/handlers/middleware_test.go:27]: Changer ProductHandler pour GameHandler ou PlayerHandler

File: pkg/handlers/middleware.go [hunk 6]
```diff
@@ -13,7 +13,7 @@ import (
 // We want our validation errors to have a standard format
 
 // MiddlewareProductValidation is used to validate incoming product JSONS
-func (productHandler *ProductsHandler) MiddlewareProductValidation(next http.Handler) http.Handler {
```
Comment (MaximeStOnge) [inline pkg/handlers/middleware.go:16]: Chnager le Product dans MiddlewareProductValidation et la variable product

File: pkg/handlers/handlers_test.go [hunk 7]
```diff
@@ -6,99 +6,17 @@ import (
...
-	}
-	if !strings.Contains(response.Body.String(), "Product not found") {
-		t.Error("Expected response : Product not found")
-	}
-}
-
-func TestAddProduct(t *testing.T) {
+func TestNewPlayer(t *testing.T) {
 	// Creating request body
 	body := &data.Product{
```
Comment (MaximeStOnge) [inline pkg/handlers/handlers_test.go:21]: Changer les Product pour Game ou Player

File: pkg/data/products.go [hunk 8]
```diff
@@ -53,8 +53,8 @@ func UpdateProduct(product *Product) error {
 	return nil
 }
 
-// AddProduct creates a new product
-func AddProduct(product *Product) {
+// NewPlayer creates a new product
+func NewPlayer(product *Product) {
```
Comment (MaximeStOnge) [inline pkg/data/products.go:57]: Changer param product pour Player

File: cmd/microservice-dispatcher/main.go [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/Ubivius/microservice-template/pkg/handlers"
+	"github.com/gorilla/mux"
+)
+
+func main() {
+	// Logger
+	logger := log.New(os.Stdout, "Template", log.LstdFlags)
+
+	// Creating handlers
+	productHandler := handlers.NewGameHandler(logger)
```
Comment (MaximeStOnge) [inline cmd/microservice-dispatcher/main.go:20]: productHandler pour GameHandler

**Review by MaximeStOnge — CHANGES_REQUESTED** [review]

**Comment by ArcticWalrus** [comment]
This branch will be used as a base for the 2 tasks in the sprint, but won't be merged now

**Pull request closed without merging**